core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -33,7 +33,7 @@
 public class ReachableFileUtil {
 
   private static final Logger LOG = LoggerFactory.getLogger(ReachableFileUtil.class);
-  private static final String METADATA_FOLDER_NAME = "metadata";
+  private static final String METADATA_FOLDER_NAME = "/metadata";
```

Comment:
I don't think we can just change this

Reply:
You are right, I changed the place where this path is constructed in `versionHintLocation()` by adding the slash there.